Source/Core/Common/ArmCPUDetect.cpp

```diff
@@ -59,82 +219,100 @@ CPUInfo::CPUInfo()
   Detect();
 }
 
-// Detects the various CPU features
 void CPUInfo::Detect()
 {
-  // Set some defaults here
-  HTT = false;
-  OS64bit = true;
-  CPU64bit = true;
-  Mode64bit = true;
   vendor = CPUVendor::ARM;
   bFMA = true;
   bFlushToZero = true;
-  bAFP = false;
+
+  num_cores = std::max(static_cast<int>(std::thread::hardware_concurrency()), 1);
 
 #ifdef __APPLE__
-  num_cores = std::thread::hardware_concurrency();
+  SysctlByName(&model_name, "machdep.cpu.brand_string");
 
   // M-series CPUs have all of these
-  bFP = true;
-  bASIMD = true;
+  // Apparently the world has accepted that these can be assumed supported "for all time".
+  // see https://github.com/golang/go/issues/42747
   bAES = true;
   bSHA1 = true;
   bSHA2 = true;
   bCRC32 = true;
 #elif defined(_WIN32)
-  num_cores = std::thread::hardware_concurrency();
-
-  // Windows does not provide any mechanism for querying the system registers on ARMv8, unlike Linux
-  // which traps the register reads and emulates them in the kernel. There are environment variables
-  // containing some of the CPU-specific values, which we could use for a lookup table in the
-  // future. For now, assume all features are present as all known devices which are Windows-on-ARM
-  // compatible also support these extensions.
-  bFP = true;
-  bASIMD = true;
-  bAES = true;
-  bCRC32 = true;
-  bSHA1 = true;
-  bSHA2 = true;
-#else
-  // Get the information about the CPU
-  num_cores = sysconf(_SC_NPROCESSORS_CONF);
-  strncpy(cpu_string, GetCPUString().c_str(), sizeof(cpu_string));
+  // NOTE All this info is from cpu core 0 only.
```

Comment:
On heterogeneous systems, I'm not sure we can assume any ordering for cores. On both my Exonos phone and Snapdragon windows laptop, core 0 is a little Cortex A55 core that hopefully dolphin will not be scheduled on. 

Which would be correct for cpu capabilities, the little core is typically the one with less features if there is a disparity. 

But for statistics reporting, we would like to report the big core. Would it be sane to scan all cores and report all unique core types in a list?

Also, I have no idea if this ordering (little then big) is required, a common convention, or simply implementation defined.

Reply:
tbh (at least at the moment) i'm not sure it really matters, because we only check for features which are required to be implemented above certain ARMv8.x revisions. Our stats would only report the wrong data if the big/little cores implemented different ARMv8 revisions. I don't know if that's possible or not.